### PR TITLE
Add gender randomization, and some additional codegen

### PR DESCRIPTION
**Pull request by dyc3**
- add gender randomization by default when generating pokemon
- add gender info to codegen
- add more species data to codegen
- add move meta category to code gen

### CODE REVIEW

File: pokemon.go [hunk 1]
```diff
@@ -39,6 +40,8 @@ type PokemonData struct {
 	BaseStats  [6]int // base stats of a Pokemon
 	EvYield    [6]int // effort points gained when Pokemon is defeated
 	GrowthRate int
+	IsBiGender bool  // true if this Pokemon must have `Male` or `Female`, false if this Pokemon must be `Genderless`
```
Comment (adapap) [inline pokemon.go]: Why not `HasGender` or `IsGenderless`?

File: pokemon_test.go [hunk 2]
```diff
@@ -210,3 +210,21 @@ var _ = Describe("Pokemon Data", func() {
...
 	})
 })
+
+var _ = Describe("RandomGender", func() {
+	DescribeTable("Should always return one gender",
+		func(rate int, gender Gender) {
+			for i := 0; i < 10000; i++ {
+				Expect(RandomGender(uint8(rate))).To(Equal(gender))
+			}
+		},
```
Comment (adapap) [inline pokemon_test.go:220]: This feels like a convoluted way to test this functionality. At this point we're just testing if a RNG is working. We can probably do just fine with 2-3 hard-coded iterations.

**Review by adapap — APPROVED** [review]
Some thoughts

File: pokemon_test.go (hunk 2 above)
Comment (dyc3) [reply pokemon_test.go:220]: This is equivalent of hard coding 10000 iterations in a row. This guarantees that `RandomGender` holds the behavior we are expecting.

File: pokemon.go (hunk 1 above)
Comment (dyc3) [reply pokemon.go]: It makes Pokemon genderless by default, which is our previous behavior.

**Pull request merged**